flink-end-to-end-tests/test-scripts/common.sh

```diff
@@ -230,30 +230,34 @@ function start_local_zk {
     done < <(grep "^server\." "${FLINK_DIR}/conf/zoo.cfg")
 }
 
-function wait_dispatcher_running {
+function wait_rest_endpoint_up {
+  local port=$1
+  local endpoint=$2
+  local name=$3
+  local regex=$4
   # wait at most 10 seconds until the dispatcher is up
-  local QUERY_URL="${REST_PROTOCOL}://${NODENAME}:8081/taskmanagers"
+  local QUERY_URL="${REST_PROTOCOL}://${NODENAME}:${port}/${endpoint}"
```

Comment:
If `wait_rest_endpoint_up` aims at being general purpose function, the API needs to be reconsidered. `REST_PROTOCOL` can be set by `set_conf_rest_ssl`, which has nothing to do with Mesos. Probably it would be better to pass the entire query url as an argument to this function. WDYT?